index.js

```diff
@@ -58,21 +59,22 @@ async function fastifyRateLimit (fastify, settings) {
   }, settings.addHeadersOnExceeding)
 
   // Global maximum allowed requests
-  globalParams.max = ((typeof settings.max === 'number' && !isNaN(settings.max)) || typeof settings.max === 'function')
+  globalParams.max = ((typeof settings.max === 'number' && Number.isFinite(settings.max) && (settings.max = Math.trunc(settings.max)) >= 0) || typeof settings.max === 'function')
     ? settings.max
     : defaultMax
 
   // Global time window
   globalParams.timeWindow = typeof settings.timeWindow === 'string'
     ? ms.parse(settings.timeWindow)
-    : typeof settings.timeWindow === 'number' && !isNaN(settings.timeWindow)
-      ? settings.timeWindow
-      : 1000 * 60
-  globalParams.timeWindowInSeconds = Math.floor(globalParams.timeWindow / 1000)
+    : typeof settings.timeWindow === 'number' && Number.isFinite(settings.timeWindow) && settings.timeWindow >= 0
+      ? Math.trunc(settings.timeWindow)
+      : defaultTimeWindow
+
+  globalParams.timeWindowInSeconds = Math.trunc(globalParams.timeWindow / 1000)
```

Comment:
didnt we decide to ceil it?

Reply:
We decided to ceil it when we are [sending it to the user](https://github.com/fastify/fastify-rate-limit/blob/efcb43cebaed1dbade63c1a3790bc2868f778aa1/index.js#L219C46-L219C46), but I don't know if we should ceil that property as well

I actually wanted to remove `timeWindowInSeconds` because it is used nowhere in the code... I just hesitate to do it because we pass globalParams to custom stores and people might be using it

Reply:
Should I do ceil there as well? Or just remove it maybe?

Reply:
idk you are deeper in the code than I

Reply:
I would remove it because a CustomStore user can just divide the timeWindow by 1000 themselves within the constructor and do a ceil or trunc (if it ends up being a decimal) depending on what they want to do

But right now, it was `.floor` before, so it's non-breaking in that regard

Reply:
I actually find it dangerous, because it might cause desync issues when timeWindow is floored or ceild